src/sap.m/src/sap/m/ListItemBase.js

```diff
@@ -471,7 +476,8 @@ sap.ui.define(['jquery.sap.global', './library', 'sap/ui/core/Control', 'sap/ui/
 			id : this.getId() + "-selectSingle",
 			groupName : this.getListProperty("id") + "_selectGroup",
 			activeHandling : false,
-			selected : this.getSelected()
+			selected : this.getSelected(),
+			enabled : this.getEnabled()
```

Comment:
aggregated controls are created only once and they are reused for rendering. This means  if you change your property the control will rerender and the checkbox/radiobutton state will not changed. You need to update enabled property of the checkbox/radiobutton with overwriting ListBase#setSelectable.